pandas/core/arrays/interval.py

```diff
@@ -87,18 +90,34 @@
 See Also
 --------
 Index : The base pandas Index type
-Interval : A bounded slice-like interval; the elements of an IntervalIndex
+Interval : A bounded slice-like interval; the elements of an %(klass)s
 interval_range : Function to create a fixed frequency IntervalIndex
-cut, qcut : Convert arrays of continuous data into Categoricals/Series of
-            Intervals
+cut : Bin values into discrete Intervals
+qcut : Bin values into equal-sized Intervals based on rank or sample quantiles
 """
 
 
 @Appender(_interval_shared_docs['class'] % dict(
     klass="IntervalArray",
-    summary="Pandas array for interval data that are closed on the same side",
+    summary="Pandas array for interval data that are closed on the same side.",
     versionadded="0.24.0",
-    name='', extra_methods='', examples='',
+    name='',
+    extra_methods='',
+    examples=textwrap.dedent("""\
+    Examples
+    --------
+    A new ``IntervalArray`` can be constructed directly from an array-like of
+    ``Interval`` objects:
+
+    >>> pd.core.arrays.IntervalArray([pd.Interval(0, 1), pd.Interval(1, 5)])
```

Comment:
I think we should keep this out of the docs until we know if or how to expose this publicly (which will in any case not be `pd.core.arrays.IntervalArray`)

Reply:
`ci/code_checks.sh` won't check the `IntervalArray` docstring if the Examples section is missing.  I've added a TODO note to myself to modify the example once `IntervalArray` has been made public; can open an issue for it as well if need be.

The issue for making extension arrays public has a milestone of 0.24.0, so I should be able to address this prior to release, and it should just be a matter of pasting the updated call. Can remove the example though, if you'd still prefer that approach.

Reply:
ping @jorisvandenbossche : thoughts on the above?

Reply:
OK, let's decide about it / clean-up later